### PR TITLE
(MAINT) Update changelog and version for release

### PR DESCRIPTION
**Pull request by jpinsonault**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## 1.0.2
+This is a bugfix release
+
+* Fixed a bug where unrecognized config file extensions caused `Hocon.load` to return an empty
+  hash instead of an error.
+* Added an optional `:syntax` key to the `Hocon.load` method to explicitly specify the file format
+* Renamed internal usage of `name` methods to avoid overriding built in `Object#name` method
```
Comment (cprice404) [inline CHANGELOG.md]: @jpinsonault I think this should probably be a Y release?  Technically that `:syntax` change is a new feature?

Comment (jpinsonault) [inline CHANGELOG.md]: That's a good point

**Comment by jpinsonault** [comment]
@cprice404 updated the changelog/version number to reflect the Y release. Also added the Gemfile.lock to the gitignore

File: CHANGELOG.md (hunk 1 above)
Comment (cprice404) [inline CHANGELOG.md]: Changelog still says 1.0.2?

Comment (jpinsonault) [inline CHANGELOG.md]: updated

Comment (cprice404) [inline CHANGELOG.md]: still says 1.0.2

Comment (cprice404) [inline CHANGELOG.md]: derp

Comment (cprice404) [inline CHANGELOG.md]: nm github was just sucking for me, showing me an old view

Comment (jpinsonault) [inline CHANGELOG.md]: yeah sorry I did some rebasing shenanigans

**Pull request merged**